client/components/recipes/UpdateRecipePage.jsx

```diff
@@ -0,0 +1,327 @@
+/* eslint-disable no-undef */
+import React from 'react';
+import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
+import { Redirect } from 'react-router-dom';
+import getRecipeAction from '../../action/recipes/getRecipeAction';
+import updateRecipeAction from '../../action/recipes/updateRecipeAction';
+import validateRecipe from '../../validations/validateRecipe';
+
+/**
+ * @description component to handle recipe update
+ *
+ * @class UpdateRecipePage
+ *
+ * @extends {React.Component}
+ */
+class UpdateRecipePage extends React.Component {
+  /**
+   * @description Creates an instance of UpdateRecipePage.
+   *
+   * @param {Object} props
+   *
+   * @memberof UpdateRecipePage
+   */
+  constructor(props) {
+    super(props);
+    this.state = {
+      name: '',
+      image: '',
+      ingredients: '',
+      direction: '',
+      errors: {
+        name: '',
+        ingredients: '',
+        direction: '',
+        image: ''
+      },
+      redirect: false
+    };
+    this.onChange = this.onChange.bind(this);
+    this.onSubmit = this.onSubmit.bind(this);
+    this.uploadWidget = this.uploadWidget.bind(this);
+  }
+
+  /**
+ * @description component lifecycle method called on component mount
+ *
+ * @memberof UpdateRecipePage
+ *
+ * @returns {undefined}
+ */
+  componentWillMount() {
+    const { id } = this.props.match.params;
+    this.props.getRecipe(id);
+  }
+
+  /**
+ * @description component lifecycle method
+ *
+ * @param {Object} nextProps
+ *
+ * @memberof UpdateRecipePage
+ *
+ * @returns {undefined}
+ */
+  componentWillReceiveProps(nextProps) {
+    const {
+      name, image, ingredients, direction
+    } = nextProps.recipe;
+    this.setState({
+      name, image, ingredients, direction
+    });
+  }
+
+  /**
+   * @description Handle change events
+   *
+   * @param {event} event
+   *
+   * @returns {undefined}
```

Comment:
add a description of the side-effect it produces despite having no return value. See other places where you skipped the descriptions